clang/lib/CodeGen/Targets/RISCV.cpp

```diff
@@ -632,6 +633,35 @@ class RISCVTargetCodeGenInfo : public CommonCheriTargetCodeGenInfo {
 
   void setTargetAttributes(const Decl *D, llvm::GlobalValue *GV,
                            CodeGen::CodeGenModule &CGM) const override {
+    if (const auto *VD = dyn_cast_or_null<VarDecl>(D)) {
+
+      if (auto *GVar = llvm::dyn_cast<llvm::GlobalVariable>(GV)) {
+        // If the clang declaration was annotated with `cheriot_mmio(...)`, we
+        // need to propagate the attribute by translating it to a matching
+        // `cheriot_cap_import` attribute in the generated LLVM IR, so that the
+        // back-end can generate the proper import entries and correctly
+        // translate references to this global.
+
+        if (VD->hasAttr<CHERIMMIODeviceAttr>()) {
+          auto *Attr = D->getAttr<CHERIMMIODeviceAttr>();
+          llvm::cheri::GlobalCapabilityImportAttr CapImportAttr(
+              llvm::cheri::GlobalCapabilityImportAttr::MMIO,
+              Attr->getDeviceName(), Attr->getEncodedPermissions());
+          GVar->addAttribute(CapImportAttr.getAttrName(), CapImportAttr.str());
+        }
+
+        // Same for the `cheriot_shared_object(...)` attribute.
+
```

Comment:
Extraneous newline

Reply:
[4cc56ac](https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/4cc56ac7f13b779c260a497f6c2844759cb396cc) removes the empty line and the one for the other case as well. I added another one before the longer comment, which might be not the correct style as well.